test/unit/helpers/ed.js

```diff
@@ -176,4 +176,35 @@ describe('ed', () => {
 			done();
 		});
 	});
+
+	describe('hexToString', () => {
+		it('should throw error if the hex is not a string', () => {
+			const buff = new Buffer.from('ABC', 'utf8');
+			return expect(() => ed.hexToBuffer(buff)).to.throw(
+				'Argument must be a string.'
+			);
+		});
+
+		it('should throw error if the hex string has invalid characters', () => {
+			const hex =
+				'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6z';
+			return expect(() => ed.hexToBuffer(hex)).to.throw(
+				'Argument must be a valid hex string.'
+			);
+		});
+
+		it('should throw error if the hex string has invalid number of characters', () => {
+			const hex =
+				'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f0';
+			return expect(() => ed.hexToBuffer(hex)).to.throw(
+				'Argument must be a valid hex string.'
+			);
+		});
+
+		it('should return buffer for a valid hex string', () => {
+			const hex =
+				'c094ebee7ec0c50ebee32918655e089f6e1a604b83bcaa760293c61e0f18ab6f';
+			return expect(ed.hexToBuffer(hex)).to.be.an.instanceof(Buffer);
+		});
```

Comment:
Please add a test case with capital alphabets. I know we downcast the keys in upper layer, but this particular unit of code also support it as logically these are valid hex strings.